File: JavaLab/src/main/java/ch2/section2/Quiz3.java
package ch2.section2;

import java.util.Scanner;

public class Quiz3 {
    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        System.out.print("소수인지 판단할 숫자 :");
        int num = stdin.nextInt();

        //나누는 숫자..
        int divisor = 2;
        boolean isPrime = true;

        while (divisor < num){
            if(num % divisor == 0){
                isPrime = false;
                break;
            }
            divisor++;
        }
        if(num ==2 || isPrime){
            System.out.println(num + " 은 소수입니다.");
        }else {
            System.out.println(num + " 은 소수가 아닙니다.");
        }
    }
}

File: KotlinLab/KotlinLab/src/main/kotlin/ch2/section4_control/Quiz3.kt
package org.example.ch2.section4_control


fun main() {
    println("양의 정수를 입력 하세요.")

    //toIntOrNull() - 입력한 문자열을 Int 로 변형.. int 변형이 안되면 null 리턴..
    // ?: 0 - 앞부분이 null 이라면 0 값을 이용하겠다..
    val num = readln().toIntOrNull() ?: 0

    for(no in 2..num){
        if(no == 2)
            print("$no ")
        else {
            var isPrime = true
            for(divisor in 2..no/2){
                if(no % divisor == 0){
                    isPrime = false
                    break
                }
            }
            if(isPrime)
                print("$no ")
        }
    }
}